File: src/main/java/org/web3j/evm/debugger/configuration/EvmRunConfigurationOptions.java
package org.web3j.evm.debugger.configuration;

import com.intellij.execution.configurations.LocatableRunConfigurationOptions;
import com.intellij.openapi.components.StoredProperty;

public class EvmRunConfigurationOptions extends LocatableRunConfigurationOptions {

    private final StoredProperty<String> contract = string("").provideDelegate(this, "contract");
    private final StoredProperty<String> method = string("").provideDelegate(this, "method");
    private final StoredProperty<String> walletPath = string("").provideDelegate(this, "walletPath");
    private final StoredProperty<String> walletPassword = string("").provideDelegate(this, "walletPassword");
  //  private final StoredProperty<String> privateKey = string("").provideDelegate(this, "privateKey");

    /*
    public String getPrivateKey() {
        return privateKey.getValue(this);
    }
*/
    public String getContract() {
        return contract.getValue(this);
    }

    public String getMethod() {
        return method.getValue(this);
    }

    public String getWalletPath() {
        return walletPath.getValue(this);
    }

    public String getWalletPassword() {
        return walletPassword.getValue(this);
    }

    public void setContract(String value) {
        contract.setValue(this, value);
    }

    public void setMethod(String value) {
        method.setValue(this, value);
    }

    public void setWalletPath(String value) {
        walletPath.setValue(this, value);
    }

    public void setWalletPassword(String value) {
        walletPassword.setValue(this, value);
    }
/*
    public void setPrivateKey(String value){
        privateKey.setValue(this,value);
    }
    */
}

File: src/main/kotlin/org/web3j/evm/debugger/run/EvmRunConfigurationOptions.kt
/*
 * Copyright 2019 Web3 Labs Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 */
package org.web3j.evm.debugger.run

import com.intellij.execution.configurations.LocatableRunConfigurationOptions
import com.intellij.openapi.components.StoredProperty

class EvmRunConfigurationOptions : LocatableRunConfigurationOptions() {
    val contract: StoredProperty<String?> = string("").provideDelegate(this, "contract")
    val method: StoredProperty<String?> = string("").provideDelegate(this, "method")
    val walletPath: StoredProperty<String?> =
        string("").provideDelegate(this, "walletPath")
    val walletPassword: StoredProperty<String?> =
        string("").provideDelegate(this, "walletPassword")

    fun getContract(): String? {
        return contract.getValue(this)
    }

    fun getMethod(): String? {
        return method.getValue(this)
    }

    fun getWalletPath(): String? {
        return walletPath.getValue(this)
    }

    fun getWalletPassword(): String? {
        return walletPassword.getValue(this)
    }

    fun setContract(value: String?) {
        contract.setValue(this, value)
    }

    fun setMethod(value: String?) {
        method.setValue(this, value)
    }

    fun setWalletPath(value: String?) {
        walletPath.setValue(this, value)
    }

    fun setWalletPassword(value: String?) {
        walletPassword.setValue(this, value)
    }
}

